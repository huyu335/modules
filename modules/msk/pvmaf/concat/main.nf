process PVMAF_CONCAT {
    tag "$meta.id"
    label 'process_single'

    // TODO nf-core: List required Conda package(s).
    //               Software MUST be pinned to channel (i.e. "bioconda"), version (i.e. "1.10").
    //               For Conda, the build (i.e. "h9402c20_2") must be EXCLUDED to support installation on different operating systems.
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'ghcr.io/msk-access/postprocessing_variant_calls:0.2.7':
        'ghcr.io/msk-access/postprocessing_variant_calls:0.2.7' }"

    input:
    tuple val(meta), path(files)
    path(header)

    output:
    tuple val(meta), path("*.maf"), emit: maf
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: '-sep "tsv"'
    def prefix = task.ext.prefix != null ? "${task.ext.prefix}" : (meta.patient != null ? "${meta.patient}" : "")
    def flagFiles = files.collect { "-f $it" }.join(' ')
    def output = prefix ? "${prefix}_combined.maf": 'multi_sample.maf'
    """
    pv maf concat \\
        $flagFiles \\
        $header \\
        --output $output \\
        $args


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pv: \$( pv --version )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: '-sep "tsv"'
    def prefix = task.ext.prefix != null ? "${task.ext.prefix}" : (meta.patient != null ? "${meta.patient}" : "")
    def flagFiles = files.collect { "-f $it" }.join(' ')
    def output = prefix ? "${prefix}_combined.maf": 'multi_sample.maf'
    """
    touch $output

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pv: \$( pv --version )
    END_VERSIONS
    """
}
