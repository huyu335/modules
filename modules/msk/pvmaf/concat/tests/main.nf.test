// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test pvmaf/concat
nextflow_process {

    name "Test Process PVMAF_CONCAT"
    script "../main.nf"
    process "PVMAF_CONCAT"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "pvmaf"
    tag "pvmaf/concat"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("maf - msk") {

        // TODO nf-core: If you are created a test for a chained module
        // (the module requires running more than one process to generate the required output)
        // add the 'setup' method here.
        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                            [ id:'chr22', patient:'test'], // meta map
                            file(params.test_data_msk['chr22']['chr22maf'], checkIfExists: true),
                            file(params.test_data_msk['chr22']['chr22maf'], checkIfExists: true)
                        ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                
                input[0] = [
                            [ id:'chr22', patient:'test'], // meta map
                            file(params.test_data_msk['chr22']['chr22maf'], checkIfExists: true),
                            file(params.test_data_msk['chr22']['chr22maf'], checkIfExists: true)
                        ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
