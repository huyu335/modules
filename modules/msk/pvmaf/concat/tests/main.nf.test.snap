{
    "chr22maf - msk - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_combined.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,51f138945e3892c0811ec5e2200e378d"
                ],
                "maf": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_combined.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,51f138945e3892c0811ec5e2200e378d"
                ]
            }
        ],
        "timestamp": "2024-02-22T11:14:02.022545"
    },
    "chr22maf - msk": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_combined.maf:md5,7b36c0fda7ebb28b27ddf51b7d2a09e8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,51f138945e3892c0811ec5e2200e378d"
                ],
                "maf": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_combined.maf:md5,7b36c0fda7ebb28b27ddf51b7d2a09e8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,51f138945e3892c0811ec5e2200e378d"
                ]
            }
        ],
        "timestamp": "2024-02-22T11:13:53.795133"
    }
}